### DISCUSSION TITLE
Should the default architecture be based on StarHub?

### DISCUSSION THREAD

**Discussion by schote** [schote/nexus-console#69] [General]
StarHub is an additional module that helps syncronize multiple measurement cards and is required if you want to use more than two cards. Our current architecture only has two measurement cards and can be synchronized without StarHub. 
As the StarHub module is not too expensive but requires major changes in implementation, we started a discussion about implementing the console software with StarHub by default.

The following points/problems have triggered this discussion:
- We want to support the use of multiple receive cards.
- Currently, all the waveforms are generated by only one AWG card. To have different sampling rates for RF and gradients, we would need separate AWG cards (e.g. high-voltage and low-voltage). Also the sequence calculation and execution is currently performed sequentially. Reducing the gradient sampling rate could significantly decrease the computation time and would lower the priority of parallelization.
- We found some problems in B0 mapping due to phase instabilities (to be investigated). The StarHub module might help to overcome these problems.

Any argument for or against the use of StarHub by default is appreciated.

**Comment by Toreil** [comment]
I think it makes sense, even if its just to avoid a fork down the line.

**Reply by schote** [reply]
I think the best way to go is starting a new branch which implements the use of starhub next to the current approach. The [pTx_implementation](https://github.com/schote/nexus-console/tree/pTx_implementation) branch already implemented the use of StarHub.
